source/code-snippets/usage-examples/findOne.ts

```diff
@@ -0,0 +1,40 @@
+import { FindOptions, MongoClient } from "mongodb";
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
+
+const client = new MongoClient(uri);
+
+type imdb = {
+  rating: number;
+  votes: number;
+  id: number;
+};
+
+interface MovieSchema extends Document {
+  title: string;
+  imdb: imdb;
+}
+
+async function run(): Promise<void> {
+  try {
+    await client.connect();
+
+    const database = client.db("sample_mflix");
+    const movies = database.collection<MovieSchema>("movies");
+    const query: object = { title: "The Room" };
+
+    const options: FindOptions = {
+      sort: { rating: -1 },
+      projection: { _id: 0, title: 1, imdb: 1 },
+    };
+
+    const movie = await movies.findOne(query, options);
```

Comment:
```suggestion
    const movie = await movies.findOne({ title: "The Room" }, {
      sort: { rating: -1 },
      projection: { _id: 0, title: 1, imdb: 1 },
    });
```

I think we should inline the query and options in the TS example because it is how user's get type hinting working without much intervention (like importing FindOptions). Copying and pasting this ^ example and playing with the usage will reveal that `title` is forced to be a `string`. Or even that `title` can map to one of the many query operators.

Reply:
Sounds good, I'll try to use inline queries as a means to get TS features working without much code intervention in future usage example rewrites.